src/server/server.ts

```diff
@@ -242,6 +278,130 @@ export class CTProtoServer<AuthRequestPayload, AuthData, ApiRequest extends NewM
     }
   }
 
+  /**
+   * Process buffer message.
+   *
+   * @param client - connected client
+   * @param message - accepted message
+   */
+  private async handleBufferMessage(client: Client<AuthData, ApiResponse, ApiUpdate>, message: Buffer): Promise<void> {
+    /**
+     * Parsing meta data from buffer message
+     */
+    const fileIdSlice = 10;
+    const chunkNumberOffset = 10;
+    const sizeOffset = 14;
+
+    const fileId = message.slice(0, fileIdSlice).toString();
+    const chunkNumber = message.readInt32BE(chunkNumberOffset);
+    const size = message.readInt32BE(sizeOffset);
+
+    /**
+     * Getting file data
+     */
+    const dataSlice = 18;
+    const fileChunk = message.slice(dataSlice, dataSlice+size);
+
+    /**
+     * Parsing payload message in buffer message
+     */
+    const strPayload = message.slice(dataSlice + size).toString();
+
+    const payload = JSON.parse(strPayload);
+
+    let file = this.uploadingFiles.find((req) => req.id === fileId);
+
+    /**
+     * Meta data of the first chunk includes additional information
+     */
+    if ( chunkNumber === 0 ) {
+      /**
+       * Create Buffer for file data
+       */
+      const fileData = Buffer.alloc(payload.fileSize, 0);
+
+      fileChunk.copy(fileData, 0);
+
+      /**
+       * Create new file object
+       */
+      this.uploadingFiles.push( { id: fileId,
+        uploadedChunksCount: 1,
+        file: fileData,
+        chunks: payload.chunks,
+        payload: payload.payload,
+        type: payload.type,
+        bufferLimit: size,
+      } );
+
+      file = this.uploadingFiles.find((req) => req.id === fileId);
+    } else {
+      if (file) {
+        /**
+         * Push file data
+         */
+        fileChunk.copy(file.file, file.bufferLimit * chunkNumber);
+
+        file.uploadedChunksCount = file.uploadedChunksCount + 1;
+      }
+    }
+
+    /**
+     * Check if chunks in uploading file objects consists, and then check fullness
+     */
+    if (file?.chunks) {
```

Comment:
please, don't use `?` here. Add statement for the `file` variable and describe the case, when it could be undefined